s2spy/time.py

```diff
@@ -304,3 +296,237 @@ def show(self) -> pd.DataFrame:
+        year_intervals = pd.Series(year_intervals[::-1], name=str(year))
+        year_intervals.index.name = "i_interval"
+        return year_intervals
+
+    def _concatenate_periods(self, year, list_periods, total_length, is_target):
+        # calculate start and end date
+        if is_target:
+            # anchor date is included
+            start_date = self._get_anchor(year)
+            end_date = start_date + pd.Timedelta(total_length, unit="D")
+            # generate date ranges
+            date_range = pd.date_range(start=start_date, end=end_date)
+        else:
+            # anchor date is excluded
+            start_date = self._get_anchor(year) - pd.Timedelta(
+                total_length + 1, unit="D"
+            )
+            end_date = self._get_anchor(year) - pd.Timedelta(1, unit="D")
+            # generate date ranges
+            date_range = pd.date_range(start=start_date, end=end_date)
+            date_range = date_range[::-1]
+        # generate intervals based on the building blocks
+        intervals = []
+        # pointer to the index of date for right boundary of interval
+        left_date_index = 0
+        # loop through all the building blocks to
+        for block in list_periods:
+            left_date_index += block.gap
+            # pointer to the index of date for left boundary of interval
+            right_date_index = left_date_index + block.length - 1
+            if is_target:
+                intervals.append(
+                    pd.Interval(
+                        date_range[left_date_index] - pd.Timedelta(1, unit="D"), # open left close right
+                        date_range[right_date_index],
+                        closed="right",
+                    )
+                )
+            else:
+                intervals.append(
+                    pd.Interval(
+                        date_range[right_date_index] - pd.Timedelta(1, unit="D"),
+                        date_range[left_date_index],
+                        closed="right",
+                    )
+                )
+            # move pointer to new start
+            left_date_index = right_date_index + 1
+
+        return intervals
+
+    def map_years(self, start: int, end: int):
+        """Adds a start and end year mapping to the calendar.
+
+        If the start and end years are the same, the intervals for only that single
+        year are returned by calendar.get_intervals().
+
+        Args:
+            start: The first year for which the calendar will be realized
+            end: The last year for which the calendar will be realized
+
+        Returns:
+            The calendar mapped to the input start and end year.
+        """
+        if start > end:
+            raise ValueError("The start year cannot be greater than the end year")
+
+        self._first_year = start
+        self._last_year = end
+        self._mapping = "years"
+        return self
+
```

Comment:
The `map_to_data` method seems to be missing here.

Reply:
Yep, as suggested by Peter, not something perfect, but simpler and easy to use as a starting point. So I think we can have `map_to_data` in next PR. Do you agree?